worker/updateChorals.js

```diff
@@ -111,54 +106,51 @@ function startNewChorals(chorals){
   })
 }
 
-// goes through all the timeouts and sets their values to unseen
-function setChoralsToUnseen(){
-  Object.keys(TIMEOUTS).forEach((ID) => {
-    TIMEOUTS[ID].seen = false
-  })
-}
-
 // anything that was not set to seen by now is unused
-function clearUnseenChorals(){
+function clearChoral(choral){
   // clear all unseen timeouts's
-  Object.keys(TIMEOUTS).forEach((ID) => {
-    if(!TIMEOUTS[ID].seen){
-      console.log("deleting choral with choral id " + ID)
-      clearTimeout(TIMEOUTS[ID].timeout_id)
-      clearInterval(TIMEOUTS[ID].timeout_id)
-      delete TIMEOUTS[ID]
-    }
-  })
+  console.log("deleting choral with choral id " + choral.choralId)
```

Comment:
Yeah it's on purpose. Instead of marking all the ones I've seen I just remove the timeout when I see it's not in the database anymore.